packages/smart-contracts/scripts-create2/contract-setup/adminTasks.ts

```diff
@@ -11,88 +13,99 @@ export const BATCH_FEE = 3;
 export const updateChainlinkConversionPath = async (
   contract: any,
   network: string,
-  nonce: number,
   gasPrice: BigNumber,
+  version = undefined,
 ): Promise<void> => {
   const currentChainlinkAddress = await contract.chainlinkConversionPath();
-  const chainlinkConversionPathAddress = chainlinkConversionPath.getAddress(network, '0.1.0');
+  const chainlinkConversionPathAddress = chainlinkConversionPath.getAddress(network, version);
   if (currentChainlinkAddress !== chainlinkConversionPathAddress) {
-    await contract.updateConversionPathAddress(chainlinkConversionPathAddress, {
-      nonce: nonce,
+    const tx = await contract.updateConversionPathAddress(chainlinkConversionPathAddress, {
       gasPrice: gasPrice,
     });
+    await tx.wait(1);
   }
 };
 
 export const updateSwapRouter = async (
   contract: any,
   network: string,
-  nonce: number,
   gasPrice: BigNumber,
 ): Promise<void> => {
   const currentSwapRouter = await contract.swapRouter();
   if (currentSwapRouter !== uniswapV2RouterAddresses[network]) {
-    await contract.setRouter(uniswapV2RouterAddresses[network], {
-      nonce: nonce,
+    const tx = await contract.setRouter(uniswapV2RouterAddresses[network], {
       gasPrice: gasPrice,
     });
+    await tx.wait(1);
   }
 };
 
-export const updateRequestSwapFees = async (
-  contract: any,
-  nonce: number,
-  gasPrice: BigNumber,
-): Promise<void> => {
-  const currentFees = await contract.requestSwapFees();
-  if (currentFees !== REQUEST_SWAP_FEES) {
-    await contract.updateRequestSwapFees(REQUEST_SWAP_FEES, { nonce: nonce, gasPrice: gasPrice });
+export const updateRequestSwapFees = async (contract: any, gasPrice: BigNumber): Promise<void> => {
+  const currentFees: BigNumber = await contract.requestSwapFees();
+  if (!currentFees.eq(REQUEST_SWAP_FEES)) {
+    console.log(`currentFees: ${currentFees.toString()}, new fees: ${REQUEST_SWAP_FEES}`);
```

Comment:
Great to have the logs.
It might be better to put it just after the tx.wait(1), in case there is an error. If we see it si logged, it means it is really deployed